tests/Unit/IO/Test_VolumeData.cpp

```diff
@@ -127,60 +155,58 @@ SPECTRE_TEST_CASE("Unit.IO.H5.VolumeData", "[Unit][IO][H5]") {
-          observation_value * tensor_components_and_coords[0]);
-
-    CHECK(get_first(
-              volume_file.get_tensor_component(observation_id, "y-coord")) ==
-          observation_value * tensor_components_and_coords[2]);
-    CHECK(get_second(
-              volume_file.get_tensor_component(observation_id, "y-coord")) ==
-          observation_value * tensor_components_and_coords[5]);
-
-    CHECK(get_first(
-              volume_file.get_tensor_component(observation_id, "z-coord")) ==
-          observation_value * tensor_components_and_coords[3]);
-    CHECK(get_second(
-              volume_file.get_tensor_component(observation_id, "z-coord")) ==
-          observation_value * tensor_components_and_coords[3]);
-
-    CHECK(get_first(volume_file.get_tensor_component(observation_id, "T_x")) ==
-          observation_value * tensor_components_and_coords[4]);
-    CHECK(get_second(volume_file.get_tensor_component(observation_id, "T_x")) ==
-          observation_value * tensor_components_and_coords[6]);
-
-    CHECK(get_first(volume_file.get_tensor_component(observation_id, "T_y")) ==
-          observation_value * tensor_components_and_coords[5]);
-    CHECK(get_second(volume_file.get_tensor_component(observation_id, "T_y")) ==
-          observation_value * tensor_components_and_coords[4]);
-
-    CHECK(get_first(volume_file.get_tensor_component(observation_id, "T_z")) ==
-          observation_value * tensor_components_and_coords[6]);
-    CHECK(get_second(volume_file.get_tensor_component(observation_id, "T_z")) ==
-          observation_value * tensor_components_and_coords[2]);
+    const auto read_extents = volume_file.get_extents(observation_id);
+    std::vector<size_t> element_num_points(
+        boost::make_transform_iterator(read_extents.begin(),
+                                       accumulate_extents),
+        boost::make_transform_iterator(read_extents.end(), accumulate_extents));
+    const auto read_points_by_element = [&element_num_points]() {
+      std::vector<size_t> read_points(element_num_points.size());
+      read_points[0] = 0;
+      size_t index = 1;
+      while (index != element_num_points.size()) {
+        read_points[index] =
+            read_points[index - 1] + element_num_points[index - 1];
+        index++;
+      }
+      return read_points;
+    }();
+    // Given a DataVector, corresponding to contiguous data read out of a
+    // file, find the data which was written by the grid whose extents are
+    // found at position `grid_index` in the vector of extents.
+    const auto get_grid_data = [&element_num_points, &read_points_by_element](
+                                   DataVector all_data,
+                                   const size_t grid_index) {
+      DataVector result(element_num_points[grid_index]);
+      std::copy(&all_data[read_points_by_element[grid_index]],
+                &all_data[read_points_by_element[grid_index]] +
```

Comment:
I suspect clang-tidy will dislike the pointer arithmetic on this line, just a heads up :)